container-search/src/main/java/com/yahoo/search/dispatch/SearchPath.java

```diff
@@ -100,37 +102,46 @@ private SearchPath(List<NodeSelection> nodes, Integer group) {
     }
 
     private boolean isEmpty() {
-        return nodes.isEmpty() && group == null;
+        return nodes.isEmpty() && groups.isEmpty();
     }
 
-    private Group selectGroup(SearchCluster cluster) {
-        if (group != null) {
-            Optional<Group> specificGroup = cluster.group(group);
-            if (specificGroup.isPresent()) {
-                return specificGroup.get();
+    private Group selectRandomGroupWithSufficientCoverage(SearchCluster cluster, List<Integer> groupIds) {
+        for (int numRetries = 0; numRetries < groupIds.size()*2; numRetries++) {
```

Comment:
![Random numbers](https://assets.amuniversal.com/321a39e06d6401301d80001dd8b71c47)

Perhaps add the groups to a list and pick and remove at random from that list until it is empty?

Reply:
Done